composer/trainer/trainer.py

```diff
@@ -315,10 +365,32 @@ def __init__(
             schedulers=schedulers,
         )
 
-        # Configure the profiler
-        if profiler is not None:
-            self.state.profiler = profiler.initialize_object(self.state)
+        # Configure profilers if profiling is enabled
+        if profiling:
+            self.state.profiler = Profiler(state=self.state,
+                                           event_handlers=prof_event_handlers,
+                                           skip_first=prof_skip_first,
+                                           wait=prof_wait,
+                                           warmup=prof_warmup,
+                                           active=prof_active,
+                                           repeat=prof_repeat,
+                                           merged_trace_file=prof_merged_trace_file)
             self.state.callbacks.extend(self.state.profiler.event_handlers)
+            self.state.callbacks.append(DataloaderProfiler())
+            self.state.callbacks.append(
```

Comment:
Can we condition adding in the system profiler `if sys_prof_cpu or sys_prof_memory or sys_prof_disk or sys_prof_net`? I doubt that having the polling thread in the background would have much overhead, but would still not like to have it running if nothing is using it.